### PR TITLE
Register Data Asset data storage online deployment

### PR DESCRIPTION
**Pull request by TajiHarrisMicrosoft**
# Description

When the user submits a deployment with a data storage object, the sdk will register a data asset for each collection that will keep the logging information. 

work items:
https://msdata.visualstudio.com/Vienna/_workitems/edit/2185541

If an SDK is being regenerated based on a new swagger spec, a link to the pull request containing these swagger spec changes has been included above.

# All SDK Contribution checklist:
- [x] **The pull request does not introduce [breaking changes]**
- [x] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [x] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-python/blob/main/CONTRIBUTING.md).**

## General Guidelines and Best Practices
- [x] Title of the pull request is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-python/blob/main/CONTRIBUTING.md##building-and-testing)
- [ ] Pull request includes test coverage for the included changes.

### CODE REVIEW

File: sdk/ml/azure-ai-ml/azure/ai/ml/operations/_online_deployment_operations.py [hunk 1]
```diff
@@ -343,3 +346,21 @@ def _get_workspace_location(self) -> str:
...
         return LocalEndpointMode.VSCodeDevContainer if vscode_debug else LocalEndpointMode.DetachedContainer
+
+    def _register_collection_data_assets(self, deployment: OnlineDeployment) -> None:
+        for c in deployment.data_collector.collections:
+            data_name = deployment.endpoint_name + "-" + deployment.name + "-" + c
+            data_object = Data(
+                name = data_name,
+                path = deployment.data_collector.destination.path
+                if deployment.data_collector.destination and deployment.data_collector.destination.path 
+                else "azureml://datastores/workspaceblobstore/paths/modelDataCollector",
```
Comment (nancy-mejia) [inline sdk/ml/azure-ai-ml/azure/ai/ml/operations/_online_deployment_operations.py:357]: Can we make this a constant?

File: sdk/ml/azure-ai-ml/azure/ai/ml/operations/_online_deployment_operations.py [hunk 2]
```diff
@@ -343,3 +346,21 @@ def _get_workspace_location(self) -> str:
 
     def _get_local_endpoint_mode(self, vscode_debug):
         return LocalEndpointMode.VSCodeDevContainer if vscode_debug else LocalEndpointMode.DetachedContainer
+
+    def _register_collection_data_assets(self, deployment: OnlineDeployment) -> None:
+        for c in deployment.data_collector.collections:
```
Comment (nancy-mejia) [inline sdk/ml/azure-ai-ml/azure/ai/ml/operations/_online_deployment_operations.py:351]:
```suggestion
        for collection in deployment.data_collector.collections:
```

Comment (nancy-mejia) [inline sdk/ml/azure-ai-ml/azure/ai/ml/operations/_online_deployment_operations.py:351]: Just making sure we are not missing any validation but is there any case where: `deployment.data_collector` could be undefined and throw an error when we try to get the collections?

Comment (TajiHarrisMicrosoft) [reply sdk/ml/azure-ai-ml/azure/ai/ml/operations/_online_deployment_operations.py:351]: If data_collector was undefined, it wouldn't pass the logic on line 164 to get to this call so I don't believe we need any extra validation

**Review by nancy-mejia — APPROVED** [review]

**Pull request merged**